src/System.Net.HttpListener/src/System/Net/Windows/HttpListener.Windows.cs

```diff
@@ -1335,11 +1335,11 @@ internal HttpListenerContext HandleAuthentication(RequestContextBase memoryBlob,
         
         private static void FreeContext(ref HttpListenerContext httpContext, RequestContextBase memoryBlob)
         {
-            if (httpContext != null)
+            HttpListenerContext context = Interlocked.Exchange(ref httpContext, null);
```

Comment:
We do not need the interlock. We always pass local variable, which hasn't been shared with any other thread.
Is that not the case?